cloud/amazon/ec2_vpc_route_table.py

```diff
@@ -323,15 +323,24 @@ def ensure_routes(vpc_conn, route_table, route_specs, propagating_vgw_ids,
     changed = routes_to_delete or route_specs_to_create
     if changed:
         for route_spec in route_specs_to_create:
-            vpc_conn.create_route(route_table.id,
-                                  dry_run=check_mode,
-                                  **route_spec)
+            try:
+                vpc_conn.create_route(route_table.id,
+                                      dry_run=check_mode,
+                                      **route_spec)
+            except EC2ResponseError as e:
+                if e.error_code == 'DryRunOperation':
+                    pass
 
         for route in routes_to_delete:
-            vpc_conn.delete_route(route_table.id,
-                                  route.destination_cidr_block,
-                                  dry_run=check_mode)
-    return {'changed': changed}
+            try:
+                vpc_conn.delete_route(route_table.id,
+                                      route.destination_cidr_block,
+                                      dry_run=check_mode)
+            except EC2ResponseError as e:
+                if e.error_code == 'DryRunOperation':
+                    pass
+
+    return {'changed': len(changed) > 0}
```

Comment:
Probably less brittle to do `bool(changed)` here? The variable is used only for its truthiness; the fact that the underlying type is a list is just a coincidental coupling with line 323. Other functions in this module have a `changed` variable with the same truthiness semantics but different underlying type.